include/tudat/astro/basic_astro/polyhedronBodyShapeModel.h

```diff
@@ -0,0 +1,205 @@
+     * @param computeAltitudeWithSign Flag indicating whether the altitude should be computed with sign (i.e. >0 if
+     * above surface, <0 otherwise) or having always a positive value.
+     * @param justComputeDistanceToVertices Flag indicating whether the distance should be computed wrt to all the
+     * polyhedron features or wrt to just the vertices.
+     */
+    PolyhedronBodyShapeModel(
+            const Eigen::MatrixXd& verticesCoordinates,
+            const Eigen::MatrixXi& verticesDefiningEachFacet,
+            const bool computeAltitudeWithSign,
+            const bool justComputeDistanceToVertices ):
+        verticesCoordinates_( verticesCoordinates ),
+        verticesDefiningEachFacet_( verticesDefiningEachFacet ),
+        computeAltitudeWithSign_( computeAltitudeWithSign ),
+        justComputeDistanceToVertices_( justComputeDistanceToVertices ),
+        averageRadius_( TUDAT_NAN )
+    {
+        // Check if provided settings are valid
+        basic_mathematics::checkValidityOfPolyhedronSettings( verticesCoordinates, verticesDefiningEachFacet );
+
+        // If necessary, get list with vertices defining each edge
+        if ( !justComputeDistanceToVertices_ )
+        {
+            computeVerticesDefiningEachEdge();
+        }
+    }
+
+    //! Destructor
+    ~PolyhedronBodyShapeModel( ){ }
+
+    //! Calculates the altitude above the polyhedron
+    /*!
+     *  Function to calculate the altitude above the polyhedron from a body fixed position.
+     *  Function computes the minimum distance to each of the polyhedron features (vertices, edges and facets); the
+     *  distance is only computed wrt to the edges and facets around the closest vertex. See Avillez (2022).
+     *  \param bodyFixedPosition Cartesian, body-fixed position of the point at which the altitude
+     *  is to be determined.
+     *  \return Altitude above the polyhedron.
+     */
+    double getAltitude( const Eigen::Vector3d& bodyFixedPosition );
+
+    /*! Function to set the average radius of the polyhedron.
+     *
+     * Function to set the average radius of the polyhedron. This function is necessary as currently no function to
+     * compute the average radius is implemented.
+     * @param averageRadius Average radius of the polyhedron.
+     */
+    void setAverageRadius( const double averageRadius )
+    {
+        averageRadius_ = averageRadius;
+    }
+
+    //! Function to return the average radius of the polyhedron.
+    /*!
+     *  Function to return the average radius of the polyhedron.
+     *  \return Average radius of the polyhedron.
+     */
+    double getAverageRadius( )
+    {
+        if ( std::isnan( averageRadius_ ) )
+        {
+            throw std::runtime_error( "The average radius of the polyhedron shape model was not defined." );
+        }
+
+        std::cerr << "Warning: the returned polyhedron average radius was manually defined and does not corresponds to"
```

Comment:
Would it be feasible to define a (potentially sub-optimal) algorithm to determine this from the vertices? Even if the algorithm is simplified and does not perfectly give the average radius, it would still be more robust than a manual value (?)

Reply:
There is already a function to compute the volume, so it's just R = (3V/(4pi))^{1/3} (based on Orbital Motion in
Strongly Perturbed Environments, Scheeres, 2012)... hadn't thought about that before. Done in https://github.com/MiguelAvillez/tudat/commit/72362eb0fe2378819dc1eebec57cc97ce6e07233

Reply:
Nice!